controller/controller_test.go

```diff
@@ -373,17 +401,89 @@ func TestControllerMutation(t *testing.T) {
 					Type:                  "LoadBalancer",
 					LoadBalancerIP:        "3.4.5.6",
 					ExternalTrafficPolicy: "Local",
+					ClusterIP:             "1.2.3.4",
 				},
 			},
 			want: &v1.Service{
 				Spec: v1.ServiceSpec{
 					Type:                  "LoadBalancer",
 					LoadBalancerIP:        "3.4.5.6",
 					ExternalTrafficPolicy: "Local",
+					ClusterIP:             "1.2.3.4",
 				},
 				Status: statusAssigned("3.4.5.6"),
 			},
 		},
+
+		{
+			desc: "No ClusterIP",
+			in: &v1.Service{
+				Spec: v1.ServiceSpec{
+					Type: "LoadBalancer",
+				},
+			},
+			wantErr: false,
+		},
+
+		{
+			desc: "request IP from wrong ip-family (ipv4)",
+			in: &v1.Service{
+				Spec: v1.ServiceSpec{
+					Type:           "LoadBalancer",
+					LoadBalancerIP: "1.2.3.1",
+					ClusterIP:      "3000::1",
+				},
+			},
+			wantErr: true,
```

Comment:
This didn't make sense to me, but then I re-read the code, and wantErr means something completely different than what I thought! I thought it meant "there was a transient error, the k8s control logic should retry". But in fact it means "we logged a warning event to complain that our goal is impossible to satisfy", which is exactly correct for this state.

So, disregard my past comments about "this should not error", I was wrong :)